### PR TITLE
Fix some issues/improvements

### PR DESCRIPTION
**Pull request by izigibran**
- Option to pick only defined keys 

- Use omit-deep (modified for empty object as a result to omit)

- Fixes some warnings for mongo versions

- Option to generate histories only if object change has user, reason

### CODE REVIEW

**Comment by mimani** [comment]
@izigibran Really Thankful to you for showing the interest and fixing the issues. Will it be possible for you to fix the UT as well once you are done with these fixes, will be more than glad to publish the changes made by you. Thanks again.

**Comment by zjr** [comment]
Hey @mimani, just FYI, @izigibran & are working on this together to make the mongoose-diff-history work for some special cases.

**Comment by zjr** [comment]
@mimani I would like to eliminate some duplicate stringify/parse operations on here but otherwise I think it looks good. Would you like to go over or offer any feedback before a merge?

**Comment by zjr** [comment]
@mimani this looks good to me, want to give you a bit more time to make some comments if you want.

File: package.json [hunk 1]
```diff
@@ -11,9 +11,11 @@
     "url": "https://github.com/mimani/mongoose-diff-history"
   },
   "dependencies": {
+    "deep-empty-object": "^1.0.5",
     "jsondiffpatch": "^0.1.38",
     "lodash.pick": "^4.4.0",
-    "omit-deep": "^0.3.0"
+    "omit-deep": "git://github.com/izigibran/omit-deep.git#master",
```
Comment (mimani) [inline package.json]: Is there any specific change in your repo of omit-deep which is why we are not using released version?

**Comment by zjr** [comment]
Yes, he added a “clean” option that removed empty objects. The repo seems dead and it’s unlikely to be merged.

**Comment by mimani** [comment]
Instead of pointing to master, can we create a tag(with proper name)/branch
and point to that, as it is possible that sometime you may commit some
other change in master, whcih starts breaking here.



Best Regards

Saurabh Kumar Mimani

**Pull request merged**